cpp/src/join/quadtree_point_in_polygon.cu

```diff
@@ -268,7 +268,7 @@ std::unique_ptr<cudf::table> quadtree_point_in_polygon(cudf::table_view const& p
                     "number of rings must be no less than number of polygons");
   CUSPATIAL_EXPECTS(poly_points_x.size() == poly_points_y.size(),
                     "numbers of vertices must be the same for both x and y columns");
-  CUSPATIAL_EXPECTS(poly_points_x.size() >= 3 * ring_offsets.size(),
+  CUSPATIAL_EXPECTS(poly_points_x.size() >= 3 * (ring_offsets.size() - 1),
```

Comment:
Why is this -1? If there are 2 rings, ring_offsets.size() would be 2 and there should be at least 6 vertices, not at least 3 vertices, right?

Reply:
This is a bug that was revealed by the `test_contains_properly.py` tests, which test a lot of single-polygons against single points and other geometries. The reason for the change is that the test was failing on very simple cases due to this precondition, tests that shouldn't have been failing. It turns out that `ring_offsets` is in `GeoArrow` format at this point, so a polygon with 1 ring with 3 vertices in it will have a `ring_offsets` = [0, 3] == length 2, but poly_points_x is only length 3, so it was `< 3 * ring_offsets.size()`. Because it is a GeoArrow structure with an extra end-offset, that should be subtracted from the length before comparing the number of vertices against the number of rings.